### PR TITLE
update dependency versions

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Some libraries are outdated

### 🤓 What?

- bring some dependencies update to date

### 🧪 Tested?

tested the various crawlers against metaphor instances.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Review by elic-eon — APPROVED** [review]
LGTM!

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "metaphor-connectors"
-version = "0.14.23"
+version = "0.14.24"
```
Comment (mars-lan) [inline pyproject.toml:3]: Bump again.

**Review by mars-lan — COMMENTED** [review]
Thanks for updating.

File: pyproject.toml (hunk 1 above)
Comment (alyiwang) [reply pyproject.toml]: updated

File: metaphor/unity_catalog/extractor.py [hunk 2]
```diff
@@ -454,11 +456,15 @@ def _process_column_lineage(self, dataset: Dataset, source_datasets: List[str]):
     def collect_query_logs(self) -> Iterator[QueryLog]:
         if self._query_log_config.lookback_days <= 0:
             return
+
         for query_info in self._api.query_history.list(
             filter_by=build_query_log_filter_by(self._query_log_config, self._api),
             include_metrics=True,
             max_results=self._query_log_config.max_results,
         ):
+            if not query_info.query_text or not query_info.user_name:
```
Comment (mars-lan) [inline metaphor/unity_catalog/extractor.py:465]: Check using `is None` is safer?

Comment (alyiwang) [reply metaphor/unity_catalog/extractor.py]: updated

**Review by mars-lan — APPROVED** [review]

**Pull request merged**